scripts/vscripts/tf2ware_ultimate/specialrounds/barrels.nut

```diff
@@ -0,0 +1,57 @@
+
+special_round <- Ware_SpecialRoundData
+({
+	name = "Barrels"
+	author = "tilderain"
+	description = "Explosive barrels will rain from the sky!"
+	category = ""
+})
+
+local barrels = []
+
+function OnPrecache()
+{
+	PrecacheModel("models/props_c17/oildrum001_explosive.mdl")
+}
+
+function OnUpdate()
+{
+	if(barrels.len() < 64)
+	{	
+		if (RandomInt(0, 48) == 0)
+			{ 
+				local plyOrigin = RandomElement(Ware_Players).GetOrigin()
+			
+				local barrel = SpawnEntityFromTableSafe("prop_physics_multiplayer",
+				{
+					//Ware_MinigameLocation.center
+					origin	 = plyOrigin + 
+								Vector(RandomFloat(-250.0, 250.0), 
+									   RandomFloat(-250.0, 250.0),
+									   RandomFloat(250.0, 1000.0))
+					model	  = "models/props_c17/oildrum001_explosive.mdl"
+				})
+			barrels.append(barrel)
+		}
+	}
+	else
+	{
+		if(barrels[0].IsValid())
+		{
+			barrels[0].TerminateScriptScope()
+			barrels[0].Kill()
+		}
+		barrels.remove(0)
+	}
+
+}
+
+function OnTakeDamage(params)
+{
+	//Barrels won't do non-self inflicted damage otherwise
+	local victim = params.const_entity
+	if(victim.GetClassname() == "prop_physics_multiplayer")
```

Comment:
Don't compare by classname as some minigames may use this classname. Give the barrel a targetname like "barrel_explosive" when spawning and then compare it with GetName here